### ISSUE TITLE
Formations and Fleet Cooperation

### ISSUE THREAD

**Issue by gchapyzh** [endless-sky/endless-sky#302]
I must have this:

>Formations

Allow making your own formations in the 'I' menu. Place where you want all your ships to be at all times. 

Hotkey that allows breaking formation for combat.

>Group Jump

Add option that allows jumping only when the entire fleet is ready.

**Comment by airtroop392** [comment]
This might be complicated without a reworking of the movement system entirely, or, unless all ships get reverse thrusters by default (and not as an optional add on).

**Comment by gchapyzh** [comment]
Maybe all ships would be designed to hold reverse thrusters, like steering?

**Comment by airtroop392** [comment]
That is sort of what I mean, yeah. Because of inertia, if you're going to want to stop a spaceship, you're going to need to exert an equal force in the opposite direction, but the problem is that in order to do so in this game, you either need reverse thrusters or you need to do a complete 180 and fire your thrusters, which takes time. In game, ships just do whatever right now in regards to formations, and even though they do kind of try to stay as a coherent grouping, normally they just fall apart and go to town on everything they see. I suppose that, if we were using this formation thing, all ships in the formation mirrored the flagship's movements perfectly, there would be no problem other than a couple of ships getting out of sync, and if they had reverse thrusters it'd work even smoother.

**Comment by gchapyzh** [comment]
@airtroop392 The positions don't have to be too perfect either, just in the general assigned area. And then ships could just perform breaking maneuvers without turning, which the game really needs right now.

**Comment by airtroop392** [comment]
Right, yeah, I have little doubts about the system being perfect, insofar as the ships stay together with tangible accuracy. And I agree that the game needs to give you more control over your fleet, as watching your interceptors crash into the enemy full force and die in literally seconds isn't always so much fun.

**Comment by gchapyzh** [comment]
My primary want for this lies in these types of situations: You want to go in fill force into a system and destroy their fleet, but you end up there first and die instantly, or your ships FTL in a horrible order, like interceptors first, then the warships. Gets quite annoying when you are trying to maintain some form of strategy. 

If you have formations, you can positions your heavies ships first, warp in ALL AT THE SAME TIME, have the heavies damage the enemy enough, and then break formation so that all the ships can engage.

On another note, fleet ships shout be allowed to move in separate wings. Each wing would have their own target rather then having your whole fleet focus on that one drone you accidentally selected. 

So, FTL in a formation, let the heavies fight the heavies, and then break formation and have your squads (My squads would probably be a heavy warship and four light warships, or something similar) fight on their own. The only ships that fight the ship you target will be your squad.

**Comment by airtroop392** [comment]
Makes sense. A structured chain of command for all ships, ships move in groups, targets for groups and preferred targeting for certain classes of ships. I originally assumed you just wanted the ships to move in formation, but this makes a lot more sense.

**Comment by gchapyzh** [comment]
That way the fleet moves in one of the two modes: Broken and Sync. In Sync mode all ships move in their assigned positions as one whole group. Each ship targets the nearest enemy to this ship's wing, so each targeted enemy receives fire from all ships in the wing. The ships, however, cant break formation and move in sync with your flagship.

Then the fleet can be ordered to Break, and each wing will become like a small fleet. All ships in the wing focus on a single ship. The flagship looses fleet command and controls only it's wing. The wings should not target ships that overwhelm them in force, and leave these targets for larger wings or until all other targets are destroyed. Then the flagship can order all ships back into Sync formation.

**Comment by gchapyzh** [comment]
![formation svg](https://cloud.githubusercontent.com/assets/8764280/11453895/83a9d98c-95d1-11e5-86d1-c8e6314fde3b.png)

**Comment by gendalv** [comment]
I'd like to have simple order selected ships to form circle or square, with empty space in the middle around a clicked point.

**Comment by tibetiroka** [comment]
This was partially resolved by #10039 and #4471.
